### PR TITLE
test: add a test for `appsody extract`

### PR DESCRIPTION
**Pull request by gireeshpunathil**
Extraction test logic:
~~Don't inspect the images, but just figure out the mount points from the debug log that is thrown out from extraction~~
 - Get the environment variables from the images that was just extracted - GetEnvVar. Switch the current directory to meet the API's needs. (refactor it so that this isn't needed)
 - For each mount that is a file, make sure the source and destination file sizes match.
 - For each mount that is a folder, make sure the source and destination folders have same content (file name match)
 - Skip mounts that were not extracted (e.g:- /mvn/...)
 - For all cases, make sure we have a `Dockerfile` extracted at the vortex of the extraction.

### CODE REVIEW

File: functest/extract_test.go [hunk 1]
```diff
@@ -0,0 +1,198 @@
...
+		// 1. Don't inspect the images, but just figure out the mount points
+		// from the debug log that is thrown out from extraction
+		// 2. For each mount that is a file, make sure the source and
+		// destination file sizes match.
+		// 3. For each mount that is a folder, make sure the source and
+		// destination folders have same content (file name match)
+		// 4. Skip mounts that were not extracted (e.g:- /mvn/...)
+		// 5. For all cases, make sure we have a Dockerfile extracted
+		// at the vortex of the extraction.
+		r, _ := regexp.Compile("APPSODY_MOUNTS Value: .*")
```
Comment (chilanti) [inline functest/extract_test.go]: What's the rationale behind using the debug output to find out what the mounts are? We like the test, but at the same time we think it's risky to depend on a debug message to figure out the mounts.

Comment (gireeshpunathil) [reply functest/extract_test.go]: @chilanti - I was concerned about this too! 

my rationale: if we want to obtain the environment of the stack, I would need to either do almost everything as `extract` do, or invoke APIs that take part in extraction logic. In ether way, any potential bug that affects `extract` can also infect into the test.

I changed it now, and took a new approach - basically run the `extract` in-process, temporarily switch the working folder the project location, get the environment through `GetEnvVar` - right now this is not exported, but a change is proposed to that effect through https://github.com/appsody/appsody/pull/322 , pls have a look!

**Comment by gireeshpunathil** [comment]
Don't land this yes - it depends on changes in https://github.com/appsody/appsody/pull/322

File: functest/extract_test.go (hunk 1 above)
Comment (chilanti) [reply functest/extract_test.go]: Thanks - will take a look - and I do see your point (it doesn't make sense to test `extract` with some of the same `extract` logic). But we'll take it.

File: functest/extract_test.go [hunk 2]
```diff
@@ -0,0 +1,208 @@
...
+		dockerFile := filepath.Join(extractDir, "Dockerfile")
+		_, err = exists(dockerFile)
+		if err != nil {
+			t.Fatal("Extraction failure, Dockerfile was not extracted into ", extractDir)
+		}
+	}
+}
+
+// private exists fn.
+// TODO: Can this be re-used from cmd/utils.go?
```
Comment (chilanti) [inline functest/extract_test.go]: Yes - we should reuse the one in utils. Please make it public.

Comment (gireeshpunathil) [reply functest/extract_test.go]: @chilanti - raised in #328 , PTAL, thanks!

**Comment by gireeshpunathil** [comment]
[ PR now blocked on #328 ]

**Pull request merged**